fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryAggregateIndexPlan.java

```diff
@@ -0,0 +1,296 @@
+    }
+
+    @Override
+    public int maxCardinality(@Nonnull RecordMetaData metaData) {
+        return indexPlan.maxCardinality(metaData);
+    }
+
+    @Override
+    public boolean isStrictlySorted() {
+        return indexPlan.isStrictlySorted();
+    }
+
+    @Override
+    public RecordQueryAggregateIndexPlan strictlySorted() {
+        return new RecordQueryAggregateIndexPlan((RecordQueryPlanWithIndex)indexPlan.strictlySorted(), recordTypeName, toRecord, partialRecordDescriptor, resultValue);
+    }
+
+    @Nonnull
+    @Override
+    public Optional<? extends MatchCandidate> getMatchCandidateMaybe() {
+        return indexPlan.getMatchCandidateMaybe();
+    }
+
+    @Nonnull
+    @Override
+    public AvailableFields getAvailableFields() {
+        return AvailableFields.NO_FIELDS;
+    }
+
+    @Override
+    public boolean hasLoadBykeys() {
+        return false;
+    }
+
+    @Nonnull
+    @Override
+    public Value getResultValue() {
+        return resultValue;
+    }
+
+    @Nonnull
+    @Override
+    public String toString() {
+        return "AggregateIndexScan(" + indexPlan + " -> " + toRecord + ")";
+    }
+
+    @Nonnull
+    @Override
+    public Set<CorrelationIdentifier> getCorrelatedTo() {
+        final var result = ImmutableSet.<CorrelationIdentifier>builder();
+        result.addAll(indexPlan.getCorrelatedTo());
+        result.addAll(resultValue.getCorrelatedTo());
+        return result.build();
+    }
+
+    @Nonnull
+    @Override
+    @SuppressWarnings("PMD.CompareObjectsWithEquals")
+    public RecordQueryAggregateIndexPlan translateCorrelations(@Nonnull final TranslationMap translationMap, @Nonnull final List<? extends Quantifier> translatedQuantifiers) {
+        final var translatedIndexPlan = indexPlan.translateCorrelations(translationMap, translatedQuantifiers);
+        final var maybeNewResult = resultValue.translateCorrelations(translationMap);
+        if (translatedIndexPlan != indexPlan || maybeNewResult != resultValue) {
+            return new RecordQueryAggregateIndexPlan(translatedIndexPlan, recordTypeName, toRecord, partialRecordDescriptor, maybeNewResult);
+        }
+        return this;
+    }
+
+    @Override
+    @SuppressWarnings("PMD.CompareObjectsWithEquals")
+    public boolean equalsWithoutChildren(@Nonnull RelationalExpression otherExpression,
+                                         @Nonnull final AliasMap equivalencesMap) {
+        if (this == otherExpression) {
+            return true;
+        }
+        if (getClass() != otherExpression.getClass()) {
+            return false;
+        }
+        final RecordQueryAggregateIndexPlan other = (RecordQueryAggregateIndexPlan) otherExpression;
+        return indexPlan.structuralEquals(other.indexPlan, equivalencesMap) &&
+               recordTypeName.equals(other.recordTypeName) &&
+               toRecord.equals(other.toRecord) &&
+               resultValue.equalsWithoutChildren(other.resultValue, equivalencesMap);
```

Comment:
should be `resultValue.semanticEquals()`

Reply:
done.